extensions/logging/log4j2_extensions/src/main/java/io/opentelemetry/contrib/logging/log4j2/OpenTelemetryJsonLayout.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.logging.log4j2;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.core.time.Instant;
+import org.apache.logging.log4j.core.util.StringBuilderWriter;
+import org.apache.logging.log4j.spi.StandardLevel;
+import org.apache.logging.log4j.util.ReadOnlyStringMap;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * This class implements a JSON layout for Log4j 2.x that will include request correlation
+ * information in the form of traceid, spanid, and traceflags fields. The format is provisional, but
+ * is designed to mirror the <a
+ * href="https://github.com/open-telemetry/oteps/blob/master/text/logs/0097-log-data-model.md">Log
+ * Data Model</a>
+ */
+@Plugin(name = "OpenTelemetryJsonLayout", category = "Core", elementType = "layout")
+public class OpenTelemetryJsonLayout extends AbstractStringLayout {
```

Comment:
Can this PR be split between OpenTelemetryJsonLayout and TraceContextDataProvider?

Reply:
I can. The OpenTelemetryJsonLayout depends on the TraceContextDataProvider. Would it be better to submit and merge the context data provider and then submit the other, or would it be better to submit the layout against the data provider branch?